paper_framework_sources/p2079_system_execution_context.md

```diff
@@ -337,6 +390,61 @@ This would work at task granularity, for each `schedule()` call that we connect
+* Uses preallocated storage on the host side, so that the default implementation doesn't need to allocate memory on the heap when adding new work to `system_scheduler`.
+* Guarantees a lifetime of at least the duration of `main()`.
+* As the default implementation is created outside of the host part, it can be shared between multiple binaries in the same process.
+* IN PROGRESS: Provides an example on implementing the `drive()` behavior by replacing the default implementation.
+* TODO: Use OS scheduler for implementation.
+
+## Addressing received feedback
+
+### Allow for system context to borrow threads
+Early feedback on the paper from Sean Parent suggested a need for the system context to carry no threads of its own and to take over the main thread.
+While in [@P2079R2] we proposed `execute_chunk` and `execute_all`, these enforce a particular implementation on the underlying execution context.
+Instead, we simplify the proposal by removing this functionality and assuming that it is implemented by link-time or compile-time replacement of the context.
+We assume that the underlying mechanism to drive the context, should one be necessary, is implementation-defined.
+This allows for custom hooks into an OS thread pool, or a simple `drive()` method in main.
+
+### Allow implementations to use Grand Central Dispatch and Windows Thread Pool
+
+In the current form of the paper, we allow implementations to define the best choice for implementing the system context for a particular system.
+This includes using Grand Central Dispatch on Apple platforms and Windows Thread Pool on Windows.
+
+In addition, we encourage implementations to allow the replaceability of the system context implementation.
+This means that users should be allowed to write their own system context implementations that depend on OS facilities.
+
+### Priorities and elastic pools
+
+Feedback from Sean Parent:
+> There is so much in that proposal that is not specified. What requirements are placed on the system scheduler? Most system schedulers support priorities and are elastic (i.e., blocking in the system thread pool will spin up additional threads to some limit).
+
+The lack of details in the specification is intentional, allowing implementers to make the best compromises for each platform.
+As different platforms have different needs, constraints, and optimization goals, the authors believe that it is in the best interest of the users to leave some of these details as Quality of Implementation (QOI) details.
+
+
+
+# Questions to ask LEWG
+
+## Do we want to standardize the "replaceability" aspect of `system_context` (as opposed to leaving this to be implementation defined)?
+
+Proposed answer: NO.
+We want to allow vendors the flexibility of choosing the best implementation.
+At the same time, the authors of this paper are willing to contribute to creating an informal basis for implementing replaceability in a common way among major vendors.
```

Comment:
We should treat an implementation that does not do this as poor QOI, especially an implementor locking us in to never being able to do that.

Reply:
I agree this would be the best approach. I was nervous about trying to define it the right way in this paper, like you a fear the delay.